unpack-trees.c

```diff
@@ -1374,8 +1378,11 @@ static int clear_ce_flags_1(struct index_state *istate,
 		int len, dtype;
 		enum pattern_match_result ret;
 
+		display_progress(istate->progress, progress_nr);
```

Comment:
We could simplify this commit if we wanted to get a little tricky.
`(cache - istate->cache)` should give the cell number which
we can cast to be the progress value.  Then we could avoid all
of the changes to the APIs.